src/components/AffectedOfferings.vue

```diff
@@ -118,6 +126,36 @@ function moduleComponentName(moduleName: string = '<module not set>', componentN
 
 defineExpose({ togglePsModuleExpansion, togglePsComponentExpansion, isExpanded });
 
+function shouldShowModuleNameEdit(moduleName: string) {
+  const expandedState = expandedModules.value[moduleName];
+  return expandedState && (!initialAffectedModules.value.includes(moduleName)
+  || affectsWithModuleName(moduleName).every(affect => !affect.uuid));
+}
+
+function addNewModule() {
+  collapseAll();
+  uniqueModuleId.value += 1;
+  const newModuleName = `New Module ${uniqueModuleId.value}`;
+  affectedModules.value.push(newModuleName);
+  expandedModules.value[newModuleName] = true;
```

Comment:
here you could just call the modified `addBlankAffect` function (outlined below, the change being to return the new blank affect) and modify the `.ps_module` property of the affect.